### PR TITLE
refactoring backwards induction

### PR DESCRIPTION
**Pull request by rm3shah**
I have refactored part of backwardsInduction in this PR by creating a function that does the calculation for eqn5 of the paper. I have also removed some unused code and added the legend for actions to the comment describing backwards induction. 

Please ensure:
- [ ] the model runs
- [ ] the values printed to the results files are the same as before the method was used (note you will see some of the results file values have changed in the diff, this is due to changing the indices in the yhi distribution on line 188 from py[i][h][y] to py[i][y][h]
- [ ] all comments removed were necessary removals 

Thanks!

### CODE REVIEW

File: model/MavenModel/MDP/src/main/java/com/team12/mdp/model.java [hunk 1]
```diff
@@ -217,6 +201,27 @@ what type of vial (z_i) you are opening in the current epoch (h)
 		}
 		return py;
 	}
+
+	public static double eqn5(int h, int t, int q1, int q2, int[] numVialDoses, int timeSlotDay, double[][] pd, 
+		double[][][] py, double[][][][] v, double[][][][] benefit, int vial, int benefitNum) {
```
Comment (brandonrhayes) [inline model/MavenModel/MDP/src/main/java/com/team12/mdp/model.java:206]: Ok, this is fine, but should probably create an object for the input instead as we have for other functions with many inputs.

**Review by brandonrhayes — CHANGES_REQUESTED** [review]
Hi Remi,

I spent some time reviewing this. I'm requesting that we review the changes to the results together. There's some discrepancies I wasn't able to determine the cause of. For some reason, the model.java file is showing more changes than I believe were actually made. This is making it hard to identify what was actually changed. I'm not sure if it's easy for you to fix this. Nice job removing about 50-lines from model.java. The legend is also great!

I've made some comments in files.

I tested your criteria:
- [x] the model runs `However, the results I'm not confident in- Although perhaps you can demonstrate something I'm missing.`
- [x] The values printed to the results files are the same as before the method was used (note you will see some of the results file values have changed in the diff, this is due to changing the indices in the yhi distribution on line 188 from py[i][h][y] to py[i][y][h] `NOTE: This does not appear to be the reason for this change. However, I was not able to address this change. We will have to look into it.`
 - [x] all comments removed were necessary removals `NOTE: As mentioned, it was hard to verify what exactly was changed in some areas, however, I did not see anything of concern.`

**Comment by rm3shah** [comment]
> Hi Remi,
> 
> I spent some time reviewing this. I'm requesting that we review the changes to the results together. There's some discrepancies I wasn't able to determine the cause of. For some reason, the model.java file is showing more changes than I believe were actually made. This is making it hard to identify what was actually changed. I'm not sure if it's easy for you to fix this. Nice job removing about 50-lines from model.java. The legend is also great!
> 
> I've made some comments in files.
> 
> I tested your criteria:
> 
> * [x]  the model runs `However, the results I'm not confident in- Although perhaps you can demonstrate something I'm missing.`
> * [x]  The values printed to the results files are the same as before the method was used (note you will see some of the results file values have changed in the diff, this is due to changing the indices in the yhi distribution on line 188 from py[i][h][y] to py[i][y][h] `NOTE: This does not appear to be the reason for this change. However, I was not able to address this change. We will have to look into it.`
> * [x]  all comments removed were necessary removals `NOTE: As mentioned, it was hard to verify what exactly was changed in some areas, however, I did not see anything of concern.`

Hi Brandon,

I have resolved all comments left here. Please take a look then squash and merge as needed. 

Thanks,
Remi

**Review by brandonrhayes — APPROVED** [review]
Looks great Remi!  All as expected now 🥇

**Pull request merged**